packages/web/src/components/splitView.tsx

```diff
@@ -43,12 +43,8 @@ export const SplitView: React.FC<SplitViewProps> = ({
   main,
 }) => {
   const defaultSize = Math.max(minSidebarSize, sidebarSize) * window.devicePixelRatio;
-  const hSetting = useSetting<number>((settingName ?? 'unused') + '.' + orientation + ':size', defaultSize);
-  const vSetting = useSetting<number>((settingName ?? 'unused') + '.' + orientation + ':size', defaultSize);
-  const hState = React.useState(defaultSize);
-  const vState = React.useState(defaultSize);
-  const [hSize, setHSize] = settingName ? hSetting : hState;
-  const [vSize, setVSize] = settingName ? vSetting : vState;
+  const [hSize, setHSize] = useSettingOrState(settingName ? `${settingName}.${orientation}:size` : undefined, defaultSize);
+  const [vSize, setVSize] = useSettingOrState(settingName ? `${settingName}.${orientation}:size` : undefined, defaultSize);
```

Comment:
There's a bug hidden in here, where both `hSize` and `vSize` will always point to the same setting. I'll write a fix in a separate PR.